torch/_higher_order_ops/associative_scan.py

```diff
@@ -37,6 +37,17 @@ def wrap_combine_fn_flat(*args, combine_fn, spec, num_leaves):
     return combined_leaves
 
 
+class AssociativeScanOp(HigherOrderOperator):
+    def __init__(self):
+        super().__init__("associative_scan")
+
+    def __call__(self, combine_fn, input, dim):
```

Comment:
We should also probably put some docstrings here about waht each of the args are. One complaint for our aten ops is that they aren't well documented. Doesn't need to happen in this PR.